packages/nextjs/components/ActivitiesItem.tsx

```diff
@@ -0,0 +1,225 @@
+      {getTargetNetwork().id === hardhat.id ? (
+        <span className={`ml-1.5 text-sm font-normal`}>
+          <Link href={blockExplorerAddressLink}>{displayAddress}</Link>
+        </span>
+      ) : (
+        <a
+          className={`ml-1.5 text-sm font-normal`}
+          target="_blank"
+          href={blockExplorerAddressLink}
+          rel="noopener noreferrer"
+        >
+          {displayAddress}
+        </a>
+      )}
+      {addressCopied ? (
+        <CheckCircleIcon
+          className="ml-1.5 text-xl font-normal text-sky-600 h-5 w-5 cursor-pointer"
+          aria-hidden="true"
+        />
+      ) : (
+        <CopyToClipboard
+          text={address}
+          onCopy={() => {
+            setAddressCopied(true);
+            setTimeout(() => {
+              setAddressCopied(false);
+            }, 800);
+          }}
+        >
+          <DocumentDuplicateIcon
+            className="ml-1.5 text-xl font-normal text-sky-600 h-5 w-5 cursor-pointer"
+            aria-hidden="true"
+          />
+        </CopyToClipboard>
+      )}
+    </div>
+  );
+};
+
+export type TActivityItemProps = {
+  address: string;
+  amount: number;
+  landedOn: string;
+};
+
+export const ActivitiesItems = ({ address, amount, landedOn }: TActivityItemProps) => {
+  const [ensAvatar, setEnsAvatar] = useState<string | null>();
+
+  const { data: fetchedEnsAvatar } = useEnsAvatar({
+    address,
+    enabled: isAddress(address ?? ""),
+    chainId: 1,
+    cacheTime: 30_000,
+  });
+
+  const blockieSizeMap = {
+    xs: 6,
+    sm: 7,
+    base: 8,
+    lg: 9,
+    xl: 10,
+    "2xl": 12,
+    "3xl": 15,
+  };
+
+  const size = "3xl";
+
+  useEffect(() => {
+    setEnsAvatar(fetchedEnsAvatar);
+  }, [fetchedEnsAvatar]);
+
+  // Skeleton UI
+  if (!address) {
+    return (
+      <div className="animate-pulse flex space-x-4">
+        <div className="rounded-md bg-slate-300 h-6 w-6"></div>
+        <div className="flex items-center space-y-6">
+          <div className="h-2 w-28 bg-slate-300 rounded"></div>
+        </div>
+      </div>
+    );
+  }
+
+  if (!ethers.utils.isAddress(address)) {
+    return <span className="text-error">Wrong address</span>;
+  }
+
+  return (
+    <>
+      <div className="flex justify-between space-x-3 px-1 my-2 mb-5">
+        <div className="  w-1/6 flex-shrink-0">
+          {ensAvatar ? (
+            // Don't want to use nextJS Image here (and adding remote patterns for the URL)
+            // eslint-disable-next-line
+          <img
+            className="p-0.5"
+            src={ensAvatar}
+            width={65}
+            height={50}
+            alt={`${address} avatar`}
+          />
+          ) : (
+            <Blockies className="p-0.5" size={blockieSizeMap[size]} seed={address.toLowerCase()} scale={4} />
+          )}
+        </div>
+        <div className="w-4/6">
+          <div className="flex space-x-10 px-1 w-full">
+            <span className="h-8"> {address?.slice(0, 7)} </span>
```

Comment:
I think it's not needed

Reply:
Alright, the decision was made based on the UI design. Due to the potential length of ens names, I opted to use the initial five characters of the user's address. However, if you believe it's better to make a different choice and we can agree on this, I'll proceed with removing it accordingly.

Reply:
remove it for now, it duplicates other address field